client/transport/src/ws/mod.rs

```diff
@@ -438,8 +439,23 @@ impl WsTransportClientBuilder {
 			&target.path_and_query,
 		);
 
-		let headers: Vec<_> =
-			self.headers.iter().map(|(key, value)| Header { name: key.as_str(), value: value.as_bytes() }).collect();
+		let headers: Vec<_> = if let Some(basic_auth) = &target.basic_auth {
+			if !self.headers.contains_key(http::header::AUTHORIZATION) {
+				let it1 =
+					self.headers.iter().map(|(key, value)| Header { name: key.as_str(), value: value.as_bytes() });
+				let it2 = std::iter::once(Header {
+					name: http::header::AUTHORIZATION.as_str(),
+					value: basic_auth.as_bytes(),
+				});
+
+				it1.chain(it2).collect()
+			} else {
+				self.headers.iter().map(|(key, value)| Header { name: key.as_str(), value: value.as_bytes() }).collect()
+			}
+		} else {
+			self.headers.iter().map(|(key, value)| Header { name: key.as_str(), value: value.as_bytes() }).collect()
+		};
```

Comment:
Maybe could remove some dupe code like this:

```suggestion
let headers = match &target.basic_auth {
    Some(basic_auth) if if !self.headers.contains_key(http::header::AUTHORIZATION) => {
        // auth header stuff
    },
    _ => self.headers.iter().map(|(key, value)| Header { name: key.as_str(), value: value.as_bytes() }).collect()
};
```